doc/running-devcontainer.md

```diff
@@ -0,0 +1,173 @@
+# Running on your machine via a Dev Container
+
+This guide describes how to run Control Panel locally with a Dev Container. This requires Docker and an IDE that can work with Dev Containers (VSCode advised). This will setup the majority of things for you, including your database, Redis, and AWS credentials.
+
+## Prerequisites
+
+### Programs
+VSCode with Python and Dev Container extensions and Docker Desktop are recommended. You will also need to be able to access AWS.
+
+### Git
+In order to get git to recognise your SSH key while working in the dev container:
+- Add your SSH keys to your ssh agent by running `ssh-add <.ssh-location>/<ssh-key-filename-without-extension>` e.g. `ssh-add ~/.ssh/id_ed25519`
+- Set your signing key in your `.gitconfig` file to the contents of your public key by running `git config --global user.signingkey key::<your-key-here>`
+
+This will allow you to use git while working in the container.
+
+## 1. Building the Dev Container
+You will need Docker Desktop along with VSCode and the Dev Container extension installed to build the container.
+
+To build the dev container, ensure docker desktop is running, then open the AP Control Panel project in VSCode. Open the command pallet by hitting command+shift+p and search for `Dev Containers: Reopen in container` and hit enter. This will build the dev container.
+
+If you are using a workspace with multiple applications, search for `Dev Containers: Open folder in Container…` instead, then select the Control Panel folder. Once the dev container has finished building, it should install all the required Python and npm dependencies, as well as run the migrations and install the helm charts.
+
+As part of the install, it will also try and create your AWS and Kubernetes config. Be sure to keep an eye on your terminal and navigate to the AWS SSO site when prompted. Once this has finished, select yes on the second prompt then your dev container will have finished building.
+
+## 2. Local Environment
+
+### Environment variables
+
+Download the copy of the working `.env` file from [1Password](https://ministryofjustice.1password.eu/vaults/skgdudwgk3ojqiwigoxrmpngle/tags/fioo45n5zohrdsf6mkdikito4d/osqkxdboemvpsgl4g2zt3kyuia). Copy the contents into a `.env` file in the root directory of the project.
```

Comment:
You will need the `.env` in place before starting the container so it will start correctly from my testing

Reply:
Moved this part to the pre requisites in the doc